apple/Tests/TestCases/Profile/ProfileTests.swift

```diff
@@ -83,4 +80,121 @@ final class ProfileTests: Test<Profile> {
 		XCTAssertTrue(jsonString.contains("encryptedSnapshot"))
 		XCTAssertTrue(jsonString.contains("version"))
 	}
+	
+	func test_json_slow_correctness() throws {
+		let (sut, json) = vector
+		let encoded = sut.profileSnapshot()
+		let decoded = try SUT(jsonBytes: json)
+		XCTAssertEqual(encoded, json)
+		XCTAssertEqual(decoded, sut)
+	}
+	
+	func test_json_fast_correctness() throws {
+		let (sut, json) = vector
+		let refBytes = sut.profileSnapshotRef()
+		try XCTAssertEqual(refBytes.take(), json)
+		let decoded = try SUT(jsonBytesReference: .init(inner: json))
+		XCTAssertEqual(decoded, sut)
+	}
+	
+	// M2 Max: Average 0.6
+	// OMITTED - too slow (6 seconds)
+	func json_encoding_slow_performance() throws {
+		let (sut, _) = vector
+		measure {
+			let _ = profileToJsonBytes(profile: sut)
+		}
+	}
+	
+	// M2 Max: Average 0.003 (200x as fast)
```

Comment:
‼️ 200x speedup vs slow.

Compared to the previous impl of `Encodable` it is a `2000x` speed up. Yes TWO THOUSAND times faster.

Reply:
200x or 2000x :))?

Reply:
200x Rust fast vs Rust slow  - 2000x vs `Swift.Encodable` solution, i.e. the `Swift.Encodable` wrapper made the Rust call take 10x the time.